File: storage-module/src/main/java/marowak/dev/entity/Equipment.java
package marowak.dev.entity;

import lombok.Builder;

import javax.persistence.*;

@Builder
@Entity
@Table(name = "equipments")
public record Equipment(
        @Id
        @GeneratedValue(generator = "equipment_seq")
        @Column(name = "equipment_id")
        Long id,

        int slotId,

        boolean equipped,

        @Column(name = "character_name")
        String characterName,

        @Column(name = "equipment_type_id")
        int equipmentTypeId,

        int upgradeLevel
) {
}

File: admin-service/models/equipment.go
package models

type Equipment struct {
	EquipmentId     int64  `gorm:"primary_key;auto_increment" json:"id"`
	Equipped        bool   `json:"equipped"`
	SlotId          int16  `json:"slotId"`
	CharacterName   string `json:"characterName"`
	EquipmentTypeId int16  `json:"equipmentTypeId"`
	UpgradeLevel    int16  `json:"upgradeLevel"`
}

func (Equipment) TableName() string {
	return "equipments"
}
